### ISSUE TITLE
Install with pip not working with Alpine Linux

### ISSUE THREAD

**Issue by robertonscjr** [cburgdorf/rusty-rlp#16]
When I try to install the rusty-rlp library on an Alpine Linux, the following error happens:

```
~ # pip3.7 install rusty-rlp
ERROR: Could not find a version that satisfies the requirement rusty-rlp (from versions: none)
ERROR: No matching distribution found for rusty-rlp
```
I need it to install web3py and pip workflow hangs exactly on this lib. Is there any way to install it manually?

Thanks!!!!

**Comment by cburgdorf** [comment]
Olá Roberto! That's unfortunate. Am I right to assume tha you are on a 32 bit system? Or maybe you are on a 64 bit system but your Python is for a 32 bit system as [it was the case in this issue](https://github.com/cburgdorf/rusty-rlp/issues/15#issuecomment-691704680)?

Unfortunately, we are only able to provide 64 bit binaries right now. But I think it should be possible to fallback to a slower implementation for non 64 bit systems. I will look into that on Monday.

**Comment by cburgdorf** [comment]
Can you run `python -c 'import sys;print("%x" % sys.maxsize, sys.maxsize > 2**32)'` and give me the output. (https://stackoverflow.com/questions/1405913/how-do-i-determine-if-my-python-shell-is-executing-in-32bit-or-64bit)

**Comment by ahmedwk** [comment]
I have the same issue. I have windows 64-bit
here is the command I typed:
```
pip install rusty-rlp
```
I get the following error message:
```
ERROR: Could not find a version that satisfies the requirement rusty-rlp (from versions: none)
ERROR: No matching distribution found for rusty-rlp
```

when I run the command you posted
```
print("%x" % sys.maxsize, sys.maxsize > 2**32)
```
I get the following:
```
7fffffff False
```

**Comment by cburgdorf** [comment]
>It seems my Windows is 64 bits, but my Python is 32 bits. I will try to uninstall 32 bit python and reinstall 64 bit Python
Thanks

Yep, that's it. That worked [for this user](https://github.com/cburgdorf/rusty-rlp/issues/15#issuecomment-691704680), too

**Comment by robertonscjr** [comment]
Hello @cburgdorf , thanks a lot for the answer!

When I ran the `python -c 'import sys;print("%x" % sys.maxsize, sys.maxsize > 2**32)'`, the output was that: `7fffffffffffffff True`. If I understand correctly, the output indicates that my Python is 64bit, correct?

Maybe it's something in Alpine? Because I tested the installation with the **python:3.7** (Ubuntu) and it worked.

**Comment by cburgdorf** [comment]
@robertonscjr yes, looks like you are on a 64 bit system but there are problems with support for Alpine. I found a related issue in the maturin issue tracker: https://github.com/PyO3/maturin/issues/154
That's the build system that we are using to create Python packages. I just skimmed over it so I'm not yet sure how it can be resolved.

**Comment by Jonney** [comment]
On another windows, I got this.
![F19C3CDB-6669-4AF8-89AA-D57D9F8AD447](https://user-images.githubusercontent.com/603073/95776724-77b8fe80-0cf7-11eb-801c-28ee7e17239d.jpeg)

**Comment by cburgdorf** [comment]
This indicates you have Python 32 bit installed on a win64 bit system.

**Comment by Jonney** [comment]
> This indicates you have Python 32 bit installed on a win64 bit system.
> […](#)

No, I'm quite sure it's Python 64-bit and win64.
I solved this problem again.
It's because of the version of Python. Not available at 3.9, It must be <=3.8

**Comment by cburgdorf** [comment]
Ah, good catch! Supporting Python 3.9 should be as easy as adding it here (and then cutting a new release)

https://github.com/cburgdorf/rusty-rlp/blob/41af02048f60a8ce4f4193a03c4b8f3a04b4491f/.github/workflows/main.yml#L45

**Comment by Abdoulrasheed** [comment]
@cburgdorf I’m using Python 3.12 and would prefer not to downgrade to <=3.8. Would it be possible to update this to support 3.12? If it’s okay, I can update the workflow file and submit a PR.
